fendermint/vm/resolver/src/pool.rs

```diff
@@ -46,6 +49,38 @@ where
     }
 }
 
+/// Tasks emitted by the pool for background resolution.
+#[derive(Clone)]
+pub struct ResolveTask {
+    /// Content to resolve.
+    key: ResolveKey,
+    /// Flag to flip when the task is done.
+    is_resolved: TVar<bool>,
```

Comment:
No, there is only up to one resolve task working on a `ResolveTask`, because of the way `own` and `theirs` are both awaited before the whole task is rescheduled for later.